stdpopsim/models.py

```diff
@@ -69,6 +69,43 @@ def verify_population_configurations_equal(
         raise UnequalModelsError("Growth rates differ")
 
 
+def sampling_times_equal(
+        populations1, populations2, rtol=DEFAULT_RTOL, atol=DEFAULT_ATOL):
+    """
+    Returns True if the sampling times for the specified lists of Populations
+    objects are equal to the specified tolerances.
+    """
+    try:
+        verify_sampling_times_equal(
+            populations1, populations2, rtol=rtol, atol=atol)
+        return True
+    except UnequalModelsError:
+        return False
+
+
+def verify_sampling_times_equal(
+        populations1, populations2, rtol=DEFAULT_RTOL, atol=DEFAULT_ATOL):
+    """
+    Check the the sampling times for lists of population objects are the same
+    """
+    # Retrieve sampling times
+    sampling_times1 = np.array([p.sampling_time for p in populations1])
+    sampling_times2 = np.array([p.sampling_time for p in populations2])
+    # Check for equal vector length
+    if len(sampling_times1) != len(sampling_times2):
+        raise UnequalModelsError("Different numbers of populations")
+    # Get indicies where None values present and compare
+    s1_none = np.where(sampling_times1 == None) # noqa
```

Comment:
I'm surprised numpy arrays can store None, but it turns out it's using dtype=object arrays. Learn something new every day!